src/Application.vala

```diff
@@ -103,12 +103,57 @@ public class Music.Application : Gtk.Application {
         settings.bind ("window-maximized", main_window, "maximized", SettingsBindFlags.SET);
     }
 
+    // FIXME(aitor-gomila): improve algorithm and make code cleaner
+    Array<File> listDirectory(string directory) {
+        var dir = Dir.open(directory, 0);
+        string? name = null;
+        Array<File> elements = new Array<File>();
```

Comment:
Why not use File[]?

Reply:
Native C arrays need to have a set size. So if you want to open a directory inside a directory, you need to get the number of files in that folder, or just use a GLib Array. Probably better to use native C arrays in this case, will check out how it's done.

Reply:
They are resizable in Vala https://wiki.gnome.org/Projects/Vala/Tutorial#Arrays

Reply:
Is there a way to get the quantity of files in a directory beforehand?